### PR TITLE
Revert codegen for broken env config feature

### PR DESCRIPTION
**Pull request by Velfi**
_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by rcoh** [comment]
This needs to target the release branch

**Comment by Velfi** [comment]
OK, targeting release branch

File: aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3ExpressDecorator.kt [hunk 1]
```diff
@@ -83,25 +80,6 @@ class S3ExpressDecorator : ClientCodegenDecorator {
...
-    override fun extraSections(codegenContext: ClientCodegenContext): List<AdHocCustomization> {
-        return listOf(
-            adhocCustomization<SdkConfigSection.CopySdkConfigToClientConfig> { section ->
-                rust(
-                    """
-                    ${section.serviceConfigBuilder}.set_disable_s3_express_session_auth(
-                        ${section.sdkConfig}
-                            .service_config()
-                            .and_then(|conf| {
-                                let str_config = conf.load_config(service_config_key("AWS_S3_DISABLE_EXPRESS_SESSION_AUTH", "s3_disable_express_session_auth"));
```
Comment (rcoh) [inline aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3ExpressDecorator.kt:96]: Do we want to disable these changes that have no global value? It seems like we could be strategic and only disable the region one. I'm fine with the larger revert though

Comment (Velfi) [reply aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3ExpressDecorator.kt:96]: Good point, I'll leave this one in.

**Review by ysaito1001 — APPROVED** [review]
The latest codegen diff LGTM

**Comment by ysaito1001** [comment]
Nit: [this](https://github.com/smithy-lang/smithy-rs/pull/3493/files#diff-722851a4ed160fa1d80699fd516f15194371ed7ab52cb80cd9add9d16d7b1e9f) can also be reverted probably, but not critical

**Pull request merged**